### PR TITLE
RET-4890: DB queue (No Merge)

### PR DESCRIPTION
**Pull request by JackReevies**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RET-4890

### Change description ###

PR not to be merged as part of RET-4890 but to be merged later when MULDB goes live

**Does this PR introduce a breaking change?** (check one with "x")

```
[] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/ethos/ecm/consumer/tasks/WorkQueueTask.java [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        }
+    }
+
+    private void setComplete(Connection conn, int id) {
+        try (CallableStatement call = conn.prepareCall("{ call complete_work(?) }")) {
+            Array sqlArray = conn.createArrayOf("int4", new Integer[] { id });
+            call.setArray(1, sqlArray);
+            call.execute();
+        } catch (SQLException ex) {
+            log.error("Failed marking work as errored", ex);
```
Comment (fergusoshea) [inline src/main/java/uk/gov/hmcts/reform/ethos/ecm/consumer/tasks/WorkQueueTask.java:149]: Should this be "Failed marking work as complete"?

**Review by fergusoshea — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/ethos/ecm/consumer/tasks/WorkQueueTask.java (hunk 1 above)
Comment (JackReevies) [reply src/main/java/uk/gov/hmcts/reform/ethos/ecm/consumer/tasks/WorkQueueTask.java:149]: Good spot, it should be

**Review by fergusoshea — APPROVED** [review]

**Review by jip1984 — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/ethos/ecm/consumer/config/CosDBConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+public class CosDBConfiguration {
+    @Value("${etcos.spring.datasource.url}")
+    private String url;
+    @Value("${etcos.spring.datasource.username}")
+    private String username;
+    @Value("${etcos.spring.datasource.password}")
+    private String password;
+
+    @Bean(name = "etcosDataSource")
+    public DataSource etcosDataSource() {
```
Comment (ss-evoco) [inline src/main/java/uk/gov/hmcts/reform/ethos/ecm/consumer/config/CosDBConfiguration.java:22]: Does it need the FlywayDataSource tag ?

File: src/main/resources/db/migration/V005__WorkTable.sql [hunk 3]
```diff
@@ -0,0 +1,9 @@
+
```
Comment (ss-evoco) [inline src/main/resources/db/migration/V005__WorkTable.sql:1]: Is this for preview ? Will flyway work with two scripts creating the same table ?

**Review by ss-evoco — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/ethos/ecm/consumer/config/CosDBConfiguration.java (hunk 2 above)
Comment (JackReevies) [reply src/main/java/uk/gov/hmcts/reform/ethos/ecm/consumer/config/CosDBConfiguration.java:22]: Deffo not, we don't want flyway operating on a database that isnt owned by message-handler. the migrations are handled by et-cos

File: src/main/resources/db/migration/V005__WorkTable.sql (hunk 3 above)
Comment (JackReevies) [reply src/main/resources/db/migration/V005__WorkTable.sql:1]: yeah preview only - its the same table in two different dbs

**Pull request closed without merging**